buildkitd/dockerd-wrapper.sh

```diff
@@ -88,7 +98,7 @@ EOF
             cat /var/log/docker.log
             echo "==== End dockerd logs ===="
             echo "If you are having trouble running docker, try using the official earthly/dind image instead"
-            exit 1
+            return 1
```

Comment:
is this intentionally? Does this mean we need to ensure `execute()` checks for the return value now?

Reply:
Yes, since exit... exits the script, and we cant retry it then. Unless we wrap the wrapper. Lets not go there.

Reply:
I'll add an exit if retries are exhausted here instead.